python/taichi/testing.py

```diff
@@ -0,0 +1,90 @@
+# The load-on-request closure (LoR-C) infrastructure, credit by @archibate
```

Comment:
<del>fun :)</del>
Because I don't want to load these function when not running in test, recall that we removed `pytest` from hard-deps.
You may say, why not:
```py
# taichi/testing.py
def archs():
  import pytest
  ...

# taichi/__init__.py
from .testing import *
```

well, but we need pytest decorators, which is evaluated on load:
```py
# taichi/testing.py
import pytest

@pytest.fixture
def archs():
  ...

# taichi/__init__.py
from .testing import *
```
So, how to delay `import pytest` to runtime? I come with the LoR-C method:
```py
# taichi/testing.py
def _():
  import pytest

  @pytest.fixture
  def archs():
    ...

  return locals()

_ = _()
ti.archs = _['archs']
```
and in `conftest.py`, simply `import taichi.testing`, we can access using `ti.archs`.

Reply:
Thanks for the explanation. I still feel like this is not easy to understand and to maintain. We need a much easier solution before this PR can be merged.